src/runtime-rs/crates/resource/src/volume/utils.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::{fs, path::Path};
+
+use anyhow::{anyhow, Context, Result};
+
+use crate::{
+    share_fs::{do_get_guest_path, do_get_host_path},
+    volume::share_fs_volume::generate_mount_path,
+};
+use kata_sys_util::eother;
+use kata_types::mount::{get_volume_mount_info, DirectVolumeMountInfo};
+
+pub const DEFAULT_VOLUME_FS_TYPE: &str = "ext4";
+pub const KATA_MOUNT_BIND_TYPE: &str = "bind";
+pub const KATA_DIRECT_VOLUME_TYPE: &str = "directvol";
+pub const KATA_VFIO_VOLUME_TYPE: &str = "vfiovol";
+pub const KATA_SPDK_VOLUME_TYPE: &str = "spdkvol";
+
+// volume mount info load infomation from mountinfo.json
+pub fn volume_mount_info(volume_path: &str) -> Result<DirectVolumeMountInfo> {
+    get_volume_mount_info(volume_path)
+}
+
+pub fn make_path_exist_ok(path: &Path) -> Result<()> {
+    if path.exists() {
+        return Ok(());
+    }
+
+    let dir = path
+        .parent()
+        .ok_or_else(|| anyhow!("failed to find parent path for {:?}", path))?;
+
+    fs::create_dir_all(dir).map_err(|e| {
+        anyhow!(
+            "failed to create rootfs dir {}: {:?}",
+            dir.to_string_lossy(),
+            e
+        )
+    })?;
+
+    Ok(())
+}
+
+pub fn get_file_name<P: AsRef<Path>>(src: P) -> Result<String> {
+    let file_name = src
+        .as_ref()
+        .file_name()
+        .map(|v| v.to_os_string())
+        .ok_or_else(|| {
+            eother!(
+                "failed to get file name of path {}",
+                src.as_ref().to_string_lossy()
+            )
+        })?
+        .into_string()
+        .map_err(|e| anyhow!("failed to convert to string {:?}", e))?;
```

Comment:
nit: maybe we can also use `.context` here

Reply:
OK, fixed

Reply:
OK